### PR TITLE
Add Servicebus auto-config, starter and sample

### PR DESCRIPTION
**Pull request by warrenzhu25**

### CODE REVIEW

File: spring-cloud-azure-samples/spring-cloud-azure-servicebus-sample/src/main/java/example/ServiceBusQueueApplication.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+import org.springframework.util.concurrent.ListenableFutureCallback;
+
+/**
+ * Spring Integration Channel Adapters for Azure Service Bus code sample.
+ *
+ * @author Warren Zhu
+ */
+@SpringBootApplication
+@EnableIntegration
+public class ServiceBusQueueApplication {
```
Comment (yaweiw) [inline spring-cloud-azure-samples/spring-cloud-azure-servicebus-sample/src/main/java/example/ServiceBusQueueApplication.java]: is @IntegrationComponentScan needed?

File: spring-cloud-azure-samples/spring-cloud-azure-servicebus-sample/src/main/java/example/ServiceBusQueueApplication.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    private static final String INPUT_CHANNEL = "input";
+    private static final String SERVICE_BUS_QUEUE_NAME = "example";
+
+    public static void main(String[] args) {
+        SpringApplication.run(ServiceBusQueueApplication.class, args);
+    }
+
+    @Bean
+    @ServiceActivator(inputChannel = OUTPUT_CHANNEL)
+    public MessageHandler messageSender(ServiceBusQueueOperation queueOperation) {
```
Comment (yaweiw) [inline spring-cloud-azure-samples/spring-cloud-azure-servicebus-sample/src/main/java/example/ServiceBusQueueApplication.java]: any autowire for queueOperation?

File: spring-cloud-azure-samples/spring-cloud-azure-servicebus-sample/src/main/java/example/WebController.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * @author Warren Zhu
+ */
+@RestController
+public class WebController {
+
+    private final ServiceBusQueueApplication.ServiceBusOutboundGateway messagingGateway;
```
Comment (yaweiw) [inline spring-cloud-azure-samples/spring-cloud-azure-servicebus-sample/src/main/java/example/WebController.java]: when is messagingGateway injected? do we need this: messagingGateway = appCtx.getBean(ServiceBusOutboundGateway.class)?

File: spring-cloud-azure-samples/spring-cloud-azure-servicebus-sample/src/main/java/example/ServiceBusQueueApplication.java (hunk 1 above)
Comment (warrenzhu25) [reply spring-cloud-azure-samples/spring-cloud-azure-servicebus-sample/src/main/java/example/ServiceBusQueueApplication.java]: No

File: spring-cloud-azure-samples/spring-cloud-azure-servicebus-sample/src/main/java/example/ServiceBusQueueApplication.java (hunk 2 above)
Comment (warrenzhu25) [reply spring-cloud-azure-samples/spring-cloud-azure-servicebus-sample/src/main/java/example/ServiceBusQueueApplication.java]: `queueOperation` is provided by auto config

File: spring-cloud-azure-samples/spring-cloud-azure-servicebus-sample/src/main/java/example/WebController.java (hunk 3 above)
Comment (warrenzhu25) [reply spring-cloud-azure-samples/spring-cloud-azure-servicebus-sample/src/main/java/example/WebController.java]: gateway is injected by `@EnableIntegration`

File: spring-cloud-azure-autoconfigure/src/main/java/com/microsoft/azure/spring/cloud/autoconfigure/servicebus/AzureServiceBusQueueAutoConfiguration.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+/**
+ * An auto-configuration for Service Bus queue
+ *
+ * @author Warren Zhu
+ */
+@Configuration
+@AutoConfigureAfter(AzureContextAutoConfiguration.class)
+@ConditionalOnClass(QueueClient.class)
+@ConditionalOnProperty("spring.cloud.azure.servicebus.namespace")
```
Comment (sophiaso) [inline spring-cloud-azure-autoconfigure/src/main/java/com/microsoft/azure/spring/cloud/autoconfigure/servicebus/AzureServiceBusQueueAutoConfiguration.java]: Why it's `spring.cloud.azure.servicebus.namespace` but not `spring.cloud.azure.servicebus`, i.e., namespace is a must?

Comment (sophiaso) [reply spring-cloud-azure-autoconfigure/src/main/java/com/microsoft/azure/spring/cloud/autoconfigure/servicebus/AzureServiceBusQueueAutoConfiguration.java]: Will more properties other than `namespace` be added?

File: spring-cloud-azure-samples/spring-cloud-azure-servicebus-sample/src/main/java/example/ServiceBusQueueApplication.java [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+        ServiceBusMessageHandler handler = new ServiceBusMessageHandler(SERVICE_BUS_QUEUE_NAME, queueOperation);
+        handler.setSendCallback(new ListenableFutureCallback<Void>() {
+            @Override
+            public void onSuccess(Void result) {
+                LOGGER.info("Message was sent successfully.");
+            }
+
+            @Override
+            public void onFailure(Throwable ex) {
+                LOGGER.info("There was an error sending the message.");
```
Comment (sophiaso) [inline spring-cloud-azure-samples/spring-cloud-azure-servicebus-sample/src/main/java/example/ServiceBusQueueApplication.java]: `info` level or others for `onFailure`?

File: spring-cloud-azure-samples/spring-cloud-azure-servicebus-sample/src/main/java/example/ServiceBusQueueApplication.java [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+/**
+ * Spring Integration Channel Adapters for Azure Service Bus code sample.
+ *
+ * @author Warren Zhu
+ */
+@SpringBootApplication
+@EnableIntegration
+public class ServiceBusQueueApplication {
+
+    private static final Log LOGGER = LogFactory.getLog(ServiceBusQueueApplication.class);
```
Comment (Incarnation-p-lee) [inline spring-cloud-azure-samples/spring-cloud-azure-servicebus-sample/src/main/java/example/ServiceBusQueueApplication.java]: have a try on `@Log`.

Comment (Incarnation-p-lee) [reply spring-cloud-azure-samples/spring-cloud-azure-servicebus-sample/src/main/java/example/ServiceBusQueueApplication.java]: `lombok`

**Comment by warrenzhu25** [comment]
Will improve message gateway in another PR

**Pull request merged**